components/blitz/src/omero/gateway/model/TableData.java

```diff
@@ -0,0 +1,175 @@
+package omero.gateway.model;
+
+import java.util.Arrays;
+
+/**
+ * A simple data 'container' for an OMERO.table
+ * 
+ * @author Dominik Lindner &nbsp;&nbsp;&nbsp;&nbsp; <a
+ *         href="mailto:d.lindner@dundee.ac.uk">d.lindner@dundee.ac.uk</a>
+ */
+public class TableData {
+
+    /** The table header */
+    private String columnNames[];
+
+    /** Column descriptions */
+    private String descriptions[];
+
+    /** The data in form data['column index']['row data'] */
+    private Object[][] data;
+
+    /** The data types of the columns */
+    private Class<?>[] types;
+
+    /**
+     * Creates a new instance
+     * 
+     * @param columnNames
+     *            The headers; can be <code>null</code>
+     * @param descriptions
+     *            Column descriptions; can be <code>null</code>
+     * @param types
+     *            The data types of the columns
+     * @param data
+     *            The data in form data['column index']['row data']
+     */
+    public TableData(String[] columnNames, String[] descriptions,
+            Class<?>[] types, Object[][] data) {
+        this.columnNames = columnNames;
+        this.descriptions = descriptions;
+        this.data = data;
+        this.types = types;
+    }
+
+    /**
+     * Get the headers
+     * 
+     * @return See above
+     */
+    public String[] getColumnNames() {
+        return columnNames;
+    }
+
+    /**
+     * Get the column descriptions
+     * 
+     * @return See above
+     */
+    public String[] getDescriptions() {
+        return descriptions;
+    }
+
+    /**
+     * Get the data
+     * 
+     * @return See above
+     */
+    public Object[][] getData() {
+        return data;
+    }
+
+    /**
+     * Get the data types
+     * 
+     * @return See above
+     */
+    public Class<?>[] getTypes() {
+        return types;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + Arrays.hashCode(columnNames);
+        result = prime * result + Arrays.hashCode(types);
+        result = prime * result + objectArrayHashCode(data, types);
+        result = prime * result + Arrays.hashCode(descriptions);
+        return result;
```

Comment:
Ah that's interesting, thanks. I used and modified the auto-generated one from eclipse. Using `hash(Object... values)` would have been easier.